src/function_classes/chebyshev_specials.py

```diff
@@ -0,0 +1,108 @@
+import torch
+import torch.distributions as D
+
+from core import FunctionClass
+
+def generate_chebyshev_coefficients(lowest_degree, highest_degree):
+    # Create a matrix to hold the coefficients, initializing with zeros
+    n = highest_degree + 1
+    coeffs = torch.zeros(n, n, dtype=torch.int32)
+    
+    # Initial conditions for T_0(x) and T_1(x)
+    coeffs[0, 0] = 1  # T_0(x) = 1
+    coeffs[1, 1] = 1  # T_1(x) = x
+    
+    # Use the recurrence relation T_{n+1}(x) = 2xT_n(x) - T_{n-1}(x)
+    skew_index = torch.arange(1, n, dtype=torch.long)
+    for i in range(2, n):
+        coeffs[i , :][skew_index] = 2 * coeffs[i-1, :][:-1]
+        coeffs[i , :] -= coeffs[i-2, :]
+    
+    return coeffs[lowest_degree:highest_degree+1]
+
+class ChebyshevKernelLinearRegression(FunctionClass):
+
+    """
+    Class for generating linear combinations of Chebyshev polynomials in given interval of degrees
+    Linear combinations are generated randomly by sampling from a normal distribution
+    """
+
+    def __init__(self, lowest_degree=3, highest_degree=11, *args, **kwargs):
+
+        self.chebyshev_coeffs = generate_chebyshev_coefficients(lowest_degree, highest_degree).float()
+        self.lowest_degree = lowest_degree
+        self.highest_degree = highest_degree
+
+        super(ChebyshevKernelLinearRegression, self).__init__(*args, **kwargs)
+
+    def _init_param_dist(self) -> D.Distribution:
+        """Produce the distribution with which to sample parameters"""
+        # Combine each polynomial randomly per sample
+        # combinations: (batch_size, 1 coefficient for each poly in chebychev_coeffs, seq_length)
+        combinations_dist = D.Normal(torch.zeros((self.x_dist.batch_shape[0], 1, self.chebyshev_coeffs.shape[0])), 1)
```

Comment:
Cleanliness: You can replace `self.x_dist.batch_shape[0]` with `self.batch_size` here